src/commands/deliver.go

```diff
@@ -0,0 +1,92 @@
+package commands
+
+import (
+	"github.com/bwmarrin/discordgo"
+	"sandwich-delivery/src/database"
+	"sandwich-delivery/src/models"
+	"strconv"
+	"time"
+)
+
+type DeliverCommand struct{}
+
+func (c DeliverCommand) getName() string {
+	return "deliver"
+}
+
+func (c DeliverCommand) getCommandData() *discordgo.ApplicationCommand {
+	return &discordgo.ApplicationCommand{Name: c.getName(), Description: "Mark an order as delivered."}
+}
+
+func (c DeliverCommand) registerGuild() string {
+	return ""
+}
+
+func (c DeliverCommand) permissionLevel() models.UserPermissionLevel {
+	return models.PermissionLevelArtist
+}
+
+func (c DeliverCommand) execute(session *discordgo.Session, event *discordgo.InteractionCreate) {
+	var order models.Order
+	resp := database.GetDB().First(&order, "assignee = ? AND status < ?", GetUser(event).ID, models.StatusDelivered)
+
+	if resp.RowsAffected == 0 {
+		session.InteractionRespond(event.Interaction, &discordgo.InteractionResponse{
+			Type: discordgo.InteractionResponseChannelMessageWithSource,
+			Data: &discordgo.InteractionResponseData{
+				Content: "You dont have an order to mark as delivered!",
+			},
+		})
+		return
+	}
+
+	if order.Status != models.StatusInTransit {
+		session.InteractionRespond(event.Interaction, &discordgo.InteractionResponse{
+			Type: discordgo.InteractionResponseChannelMessageWithSource,
+			Data: &discordgo.InteractionResponseData{
+				Content: "You can only mark orders as delivered that are in transit!",
```

Comment:
"You can only mark orders as delivered if they are in transit!" sounds better imo